File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750-minimum-length-of-string-after-deleting-similar-ends.java
class Solution {
    public int minimumLength(String s) {
        int i = 0, j = s.length() - 1;
        
    while (i < j && s.charAt(i) == s.charAt(j)) {
        char ch = s.charAt(i);
        while (i <= j && s.charAt(i) == ch)
            ++i;
        while (i < j && s.charAt(j) == ch)
            --j;
    }
    return j - i + 1;
    }
}

File: 1750-minimum-length-of-string-after-deleting-similar-ends/1750-minimum-length-of-string-after-deleting-similar-ends.cpp
class Solution {
public:
   int minimumLength(string s) {
    int i = 0, j = s.size() - 1;
    while (i < j && s[i] == s[j]) {
        auto ch = s[i];
        while (i <= j && s[i] == ch)
            ++i;
        while (i < j && s[j] == ch)
            --j;
    }
    return j - i + 1;
   }
};